src/test/java/de/qabel/desktop/QblRESTServerTest.java

```diff
@@ -0,0 +1,117 @@
+import java.net.MalformedURLException;
+import java.net.URL;
+/**
+ * Created by dax on 24.09.15.
+ */
+public class QblRESTServerTest {
+
+	private static final String DB_NAME = "QblHelloWorldModuleTest.sqlite";
+	private static final String BOB_DROP_URL = "http://localhost:6000/123456789012345678901234567890123456789012b";
+	public static final int PORT = 9696;
+	private ModuleManager moduleManager;
+	private QblRESTServer restServer;
+	private DropActor dropActor;
+	private ResourceActor resourceActor;
+	private String requestOutput;
+
+	@Before
+	public void setUp() throws Exception {
+		Persistence<String> persistence = new SQLitePersistence(DB_NAME, "qabel".toCharArray());
+		resourceActor = new ResourceActor(persistence, EventEmitter.getDefault());
+		moduleManager = new ModuleManager(EventEmitter.getDefault(), resourceActor);
+
+		Collection<DropURL> bobDropURLs = new ArrayList<>();
+		bobDropURLs.add(new DropURL(BOB_DROP_URL));
+
+		Identity alice = new Identity("Alice", null, new QblECKeyPair());
+		Identity bob = new Identity("Bob", bobDropURLs, new QblECKeyPair());
+
+		Identities identities = new Identities();
+		identities.put(alice);
+		moduleManager.getResourceActor().writeIdentities(identities.getIdentities().toArray(new Identity[0]));
+
+		Contact bobAsContactForAlice = new Contact(alice, "Bob", bobDropURLs, bob.getEcPublicKey());
+
+		Contacts contacts = new Contacts();
+		contacts.put(bobAsContactForAlice);
+		moduleManager.getResourceActor().writeContacts(contacts.getContacts().toArray(new Contact[0]));
+
+		dropActor = new DropActor(resourceActor, EventEmitter.getDefault());
+		Thread dropActorThread = new Thread(dropActor);
+		dropActor.setInterval(500);
+		dropActorThread.start();
+
+		restServer = new QblRESTServer(PORT, resourceActor, dropActor,moduleManager);
+	}
+
+	@After
+	public void tearDown() throws InterruptedException {
+		if (restServer != null) {
+			restServer.stop();
+		}
+		File persistenceTestDB = new File(DB_NAME);
+		if(persistenceTestDB.exists()) {
+			persistenceTestDB.delete();
+		}
+	}
+
+	private int sendRequest(String ressource, String method) throws IOException {
+		String address = "http://localhost:" + PORT + "/" + ressource;
+        URL url = new URL(address);
+		HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+		conn.setRequestMethod(method);
+		conn.setRequestProperty("Accept", "application/json");
+
+        BufferedReader br = new BufferedReader(new InputStreamReader(
+                (conn.getInputStream())));
+
+        StringBuffer output = new StringBuffer();
+        String out;
+        System.out.println("Output from Server .... \n");
+        while ((out = br.readLine()) != null) {
+            output.append(out);
+        }
+
+		conn.disconnect();
+		requestOutput = output.toString();
+        return conn.getResponseCode();
+	}
+
+	@org.junit.Test
+    public void testSetUp() throws Exception {
+		Assert.assertEquals(PORT, restServer.getPort());
```

Comment:
I don't know if anyone told you before, but we are not testing basic things that will result in testing the jvm. Trivial setters and getters don't need any test. So the ```getPort()```-Method is not really needed, yet. Also some of the other getters are just there because a test. Is that going to change or was it one of this awesome autogenerate features? ;)